docker/test/integration/ssl_utils/SSL_cert_utils.py

```diff
@@ -110,7 +110,7 @@ def make_ca(common_name):
     return ca_cert, ca_key
 
 
-def make_cert(common_name, ca_cert, ca_key):
+def _make_cert(common_name, ca_cert, ca_key, is_server_auth=True, is_client_auth=True):
```

Comment:
This should be changed to only allow either server or client authentication but not both.

Reply:
This was an issue when upgrading to OpenSSL 3, when both were used the SSL handshake failed, only the client or server specific auth should be added. (Sorry I didn't elaborate before, we had this discussion previously with @martinzink in person :))

Reply:
I thought the _ prefix (aka weak “internal use” indicator) should discourage anyone to use this function externally.

Reply:
Even if it is only used internally the signature allows us to have both server and client auth that should not be possible, which is also the default.

Reply:
refactored this in https://github.com/apache/nifi-minifi-cpp/pull/1574/commits/97ca6bab7beef928c28643f5d7a14e6022605016